stormtrooper/set_fit.py

```diff
@@ -1,209 +1,96 @@
-"""Zero and few-shot classification using SetFit."""
-from typing import Iterable
+from typing import Iterable, Optional
 
-import datasets
 import numpy as np
-from setfit import SetFitModel, SetFitTrainer, get_templated_dataset
+from sentence_transformers import SentenceTransformer
 from sklearn.base import BaseEstimator, ClassifierMixin
 from sklearn.exceptions import NotFittedError
+from sklearn.linear_model import LogisticRegression
+from stormtrooper._setfit import generate_synthetic_samples, finetune_contrastive
 
-__all__ = ["SetFitZeroShotClassifier", "SetFitFewShotClassifier"]
 
-
-class SetFitZeroShotClassifier(BaseEstimator, ClassifierMixin):
-    """Scikit-learn compatible zero shot classification
-    with SetFit and sentence transformers.
+class SetFitClassifier(BaseEstimator, ClassifierMixin):
+    """Zero and few-shot classifier using the SetFit technique with encoder models.
 
     Parameters
     ----------
     model_name: str, default 'sentence-transformers/all-MiniLM-L6-v2'
-        Name of sentence transformer on HuggingFace Hub.
-    sample_size: int, default 8
-        Number of training samples to generate.
+        Name of the encoder model.
     device: str, default 'cpu'
-        Indicates which device should be used for classification.
-        Models are by default run on CPU.
-
-    Attributes
-    ----------
-    classes_: array of str
-        Class names learned from the labels.
+        Device to train and run the model on.
+    classification_head: ClassifierMixin, default None
+        Classifier to use as the last step.
+        Defaults to Logistic Regression when not specified.
+    n_epochs: int, default 10
+        Number of trainig epochs.
+    batch_size: int, default 8
+        Batch size to use during training.
+    sample_size: int, default 8
```

Comment:
```suggestion
    sample_size
```
Generally I would not specify defaults or types in docs (then it is specified twice).

Reply:
I'm using Numpy style docstrings with a plugin, and all of the code was written this way because the Sphinx plugin needed it, it's unnecessary, but I think it's fine for now.